src/test/librbd/journal/test_Replay.cc

```diff
@@ -155,7 +154,7 @@ TEST_F(TestJournalReplay, AioDiscardEvent) {
                                 librbd::io::ReadResult{read_result}, 0);
   ASSERT_EQ(0, aio_comp->wait_for_complete());
   aio_comp->release();
-  if (ictx->skip_partial_discard) {
+  if (ictx->discard_granularity_bytes == 0) {
```

Comment:
Isn't this condition inverted?  Although it probably doesn't matter because 0 is passed to AioDiscardEvent constructor and the else branch is redundant now.

Reply:
Good catch! I think we need to fix both this condition (revert) and `AioDiscardEvent` (pass `ictx->discard_granularity_bytes`). Currently the test fails when running with `rbd_skip_partial_discard = false` [1].

[1] http://pulpito.ceph.com/trociny-2019-02-22_16:48:21-rbd-wip-mgolub-testing-1-distro-basic-smithi/